### PR TITLE
v0.38 changelog parse round 2

### PR DESCRIPTION
**Pull request by isaacdevlugt**
Note that for this PR, we're just sorting the entries into categories. The entries themselves can be changed later.

### CODE REVIEW

**Review by justinpickering — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -20,109 +20,104 @@
 
 * A new function `qml.registers` has been added, enabling the creation of registers, which are implemented as a dictionary of `Wires` instances.
   [(#5957)](https://github.com/PennyLaneAI/pennylane/pull/5957)
+  [(#6102)](https://github.com/PennyLaneAI/pennylane/pull/6102)
 
 <h4>Quantum arithmetic operations 🧮</h4>
```
Comment (DSGuala) [inline doc/releases/changelog-dev.md:25]: Is this just waiting for the PRs to be merged by the end of the week?

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -20,109 +20,104 @@
...
 
 <h4>Creating spin Hamiltonians 🧑‍🎨</h4>
```
Comment (DSGuala) [inline doc/releases/changelog-dev.md:27]: These too, we're expecting them by the end of the week? :crossed_fingers:

**Review by DSGuala — APPROVED** [review]
Categories and order look good to me!
Just wondering if we'll have arithmetic and spin Hamiltonians in time for the feature freeze.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-dev.md:25]: I spoke to Soran and he feels comfortable that they'll make it in :)

**Pull request merged**